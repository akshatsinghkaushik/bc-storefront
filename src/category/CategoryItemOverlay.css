.overlay {
  display: none;
  position: absolute;
  transform: translate(5.5rem, -10rem);
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
}

.overlay-add-to-cart,
.overlay-details {
  background: none;
  background-color: gray;
  color: white;
  border: none;
  padding: 1rem;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  letter-spacing: 1px;
  font-weight: 600;
}

.overlay-details {
  margin-bottom: 2rem;
}

.overlay-details:hover,
.overlay-add-to-cart:hover {
  background-color: black;
}
