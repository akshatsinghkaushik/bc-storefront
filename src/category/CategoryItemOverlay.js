import React, { useContext } from 'react';
import './CategoryItemOverlay.css';
import { Link } from 'react-router-dom';
import CartContext from '../context/CartContext';

const CategoryItemOverlay = ({ hover, hoverElem, item }) => {
  const cartContext = useContext(CartContext);

  return (
    <div
      className="overlay"
      style={
        hover && hoverElem === item.title
          ? { display: 'flex' }
          : { display: 'none' }
      }
    >
      <Link
        className="overlay-details"
        alt="VIEW DETAILS"
        to={`/product/${item.title}`}
        style={{ textDecoration: 'none', fontSize: '10px' }}
      >
        VIEW DETAILS
      </Link>
      <button
        className="overlay-add-to-cart"
        alt="ADD TO CART"
        style={{ fontSize: '10px' }}
        onClick={() => cartContext.addItemToCart(item)}
      >
        ADD TO CART
      </button>
    </div>
  );
};

export default CategoryItemOverlay;
