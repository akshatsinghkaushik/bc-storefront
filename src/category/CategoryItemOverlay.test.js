import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import CategoryItemOverlay from './CategoryItemOverlay';
import MockCartProvider from '../tests/MockCartProvider';

it('CategoryItemOverlay Renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <MockCartProvider>
          <CategoryItemOverlay
            hover={true}
            item={{
              title: 'Mashiko-Yaki Saucer',
              brand: 'Kiriko',
              price: 18,
              description:
                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.',
              image: 'mashiko-yaki-saucer.jpg',
            }}
          />
        </MockCartProvider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
