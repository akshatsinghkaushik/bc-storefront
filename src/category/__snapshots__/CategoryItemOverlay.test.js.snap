// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItemOverlay Renders correctly 1`] = `
<div
  className="overlay"
  style={
    Object {
      "display": "none",
    }
  }
>
  <a
    alt="VIEW DETAILS"
    className="overlay-details"
    href="/product/Mashiko-Yaki Saucer"
    onClick={[Function]}
    style={
      Object {
        "fontSize": "10px",
        "textDecoration": "none",
      }
    }
  >
    VIEW DETAILS
  </a>
  <button
    alt="ADD TO CART"
    className="overlay-add-to-cart"
    onClick={[Function]}
    style={
      Object {
        "fontSize": "10px",
      }
    }
  >
    ADD TO CART
  </button>
</div>
`;
